drivers/cache/cache.go

```diff
@@ -0,0 +1,244 @@
+/*
+Package cache provides a Hord database driver for a look-aside cache. To use this driver, import it as follows:
+
+	import (
+	    "github.com/madflojo/hord"
+	    "github.com/madflojo/hord/cache"
+	)
+
+# Connecting to the Database
+
+Use the Dial() function to create a new client for interacting with the cache.
+
+	// Handle database connection
+	var database hord.Database
+	...
+
+	// Handle cache connection
+	var cache hord.Database
+	...
+
+	var db hord.Database
+	db, err := cache.Dial(cache.Config{
+		Database: database,
+		Cache: 	  cache,
+	})
+	if err != nil {
+	    // Handle connection error
+	}
+
+# Initialize database
+
+Hord provides a Setup() function for preparing a database. This function is safe to execute after every Dial().
+
+	err := db.Setup()
+	if err != nil {
+	    // Handle setup error
+	}
+
+# Database Operations
+
+Hord provides a simple abstraction for working with the cache, with easy-to-use methods such as Get() and Set() to read and write values.
+
+	// Handle database connection
+	var database hord.Database
+	database, err := cassandra.Dial(cassandra.Config{})
+	if err != nil {
+		// Handle connection error
+	}
+
+	// Handle cache connection
+	var cache hord.Database
+	cache, err := redis.Dial(redis.Config{})
+	if err != nil {
+		// Handle connection error
+	}
+
+	// Connect to the Cache database
+	db, err := cache.Dial(cache.Config{
+		Database: database,
+		Cache: 	  cache,
+	})
+	if err != nil {
+	    // Handle connection error
+	}
+
+	err := db.Setup()
+	if err != nil {
+	    // Handle setup error
+	}
+
+	// Set a value
+	err = db.Set("key", []byte("value"))
+	if err != nil {
+	    // Handle error
+	}
+
+	// Retrieve a value
+	value, err := db.Get("key")
+	if err != nil {
+	    // Handle error
+	}
+*/
+package cache
+
+import (
+	"errors"
+
+	"github.com/madflojo/hord"
+)
+
+// Config provides the configuration options for the Cache driver.
+type Config struct {
+	Database hord.Database
+	Cache    hord.Database
+}
+
+// Cache is used to store data in a look-aside caching pattern. It also satisfies the Hord database interface.
+type Cache struct {
```

Comment:
Non-blocker: It may be useful to give users methods to retrieve the `data` and `cache`. In theory, they should have access to it, but if this struct instance is passed down several packages, they may not, but they may find the need to access the cache or database directly outside of the caching pattern.

It's a thought, but I wouldn't hold up the PR for it.